src/pages/blog/partial-hydration.mdx

```diff
@@ -0,0 +1,238 @@
+
+You might wonder how we would actually go about differentiating between the render modes of content writers & consumers on the server. The proper answer to this question is to use `{ draftMode } from "next/headers"`. Draft mode is a state that is managed by a server-side cookie. All pages generated by SSR/SSG/ISR will assume by default that `draftMode().isEnabled === false`. If you have cached pages through SSG/ISR but you have a `draftMode` cookie attached to your request, you will bypass the cache and get a unique server render per request. This allows you to get the fastest renders to your content consuming users, while your content writers get a fully dynamic response. The state of `draftMode().isEnabled` can be updated by running `draftMode().enable()` or `draftMode().disable()` in any server component, `route.ts` request, or server action.
+
+In the [repository for the website](https://github.com/EthanStandel/cakeelizabethdotcom) of my wife's business, [Cake Elizabeth](https://cakeelizabeth.com/), I created a series of wrapper componentst that make this process very repeatable. The model is two components exported under one title. The first component is a server component called [`LiveContentDataServer`](https://github.com/EthanStandel/cakeelizabethdotcom/blob/main/src/utils/LiveContentData/LiveContentDataServer.tsx) which takes in a component to render as well as as the content type name that the page needs. The server component checks if `draftMode().isEnabled`. If you are a regular consuming user and `draftMode().isEnabled === false` then the server component just fetches the content data that the component needs and then passes it to that component and renders it. If `draftMode().isEnabled` then everything gets passed down to a render of another component called [`LiveContentDataClient`](https://github.com/EthanStandel/cakeelizabethdotcom/blob/main/src/utils/LiveContentData/LiveContentDataClient.tsx) along with the initial render of the data. The client component takes the initial data render and passes it into a hook that is prepared to listen for subsequent data updates. That hook re-returns the initial data on the server (for the Next HTML render) and _that_ instance of the data is passed down into the component that was passed forward from `LiveContentDataServer` and all rendered together. Now, because the passed component is being rendered in `LiveContentDataClient` which is a client component, it is rendered as a client component. I then, for cleanliness add a barrel export of `export { LiveContentDataServer as LiveContentData } from "./LiveContentDataServer"` so that the model is clear that in practice, only the server component should be utilized directly and the client component should only be run from the server component.
```

Comment:
I need to re-read this paragraph tomorrow but it's a heady one.

Reply:
I tried my best to break it up a bit and I did remove the note about the barrel export. Code cleanliness was just not relevant here and only muddied up the current topic.